src/broadcast/filelistener/filelistener.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <QFile>
+#include <QThread>
+#include "preferences/dialog/dlgprefmetadata.h"
+#include "control/controlpushbutton.h"
+#include "broadcast/scrobblingservice.h"
+
+class FileListener: public ScrobblingService {
+    Q_OBJECT
+  public:
+    explicit FileListener(UserSettingsPointer pSettings);
+    ~FileListener() override;
+    void slotBroadcastCurrentTrack(TrackPointer pTrack) override;
+    void slotScrobbleTrack(TrackPointer pTrack) override;
+    void slotAllTracksPaused() override;
+  signals:
+    void deleteFile();
+    void moveFile(QString destination);
+    void writeMetadataToFile(QByteArray contents);
+    void clearFile();
+  private slots:
+    void slotFileSettingsChanged(double value);
+  private:
+
+    struct WrittenMetadata {
+        QString title,artist;
+        bool isEmpty() {
+            return title.isEmpty() && artist.isEmpty();
+        }
+    };
+
+    void updateStateFromSettings();
+    void updateFile();
+
+    ControlPushButton m_COsettingsChanged;
+    UserSettingsPointer m_pConfig;
+    FileSettings m_latestSettings;
+    QThread m_workerThread;
+    WrittenMetadata m_fileContents;
+    bool m_filePathChanged = false;
```

Comment:
We do not use initializer in class definition, this introduced a hidden static constructor.
It is also out of sight if you search the cop file.
Please move this to the normal constructor.

Reply:
Alright.